Sources/SocialAuthentication/AppleAuthentication/AppleAuthentication.swift

```diff
@@ -71,10 +66,89 @@ extension AppleAuthentication {
     }
 }
 
-// MARK: - UserId
+// MARK: - Sign Out
+
+extension AppleAuthentication {
+    
+    public func signOut() {
+        
+        deletePersistedUser()
+    }
+}
+
+// MARK: - User
+
+extension AppleAuthentication {
+    
+    public func getCurrentUserProfile() -> AppleUserProfile? {
+        
+        guard
+            let email = getUserEmail(),
+            let familyName = getUserFamilyName(),
+            let givenName = getUserGivenName()
+        else {
+            return nil
+        }
+        
+        return AppleUserProfile(
+            email: email,
+            familyName: familyName,
+            givenName: givenName
+        )
+    }
+    
+    private func deletePersistedUser() {
```

Comment:
Hey @rachaelblue what are your thoughts on encapsulating this user persistence in another class?  Doing so would reduce some of the responsibilities and functionality in this class.  Thinking from a SOLID and unit tests point of view.

Reply:
Agreed, there's a lot of logic in there between keychain/user defaults.  Went ahead and moved it all into `AppleUserPersistentStore`.